File: nas_spec/code_samples/C#/workflows@{workflowId}@actions@{workflowActionId}/put.cs
// For more information please refer to https://github.com/checkout/checkout-sdk-net
using Checkout.Workflows.Actions;
using Checkout.Workflows.Actions.Request;

//API keys
ICheckoutApi api = CheckoutSdk.Builder().StaticKeys()
    .SecretKey("secret_key")
    .Environment(Environment.Sandbox)
    .HttpClientFactory(new DefaultHttpClientFactory())
    .Build();

//OAuth
ICheckoutApi api = CheckoutSdk.Builder().OAuth()
    .ClientCredentials("client_id", "client_secret")
    .Scopes(OAuthScope.Flow)
    .Environment(Environment.Sandbox)
    .HttpClientFactory(new DefaultHttpClientFactory())
    .Build();

WorkflowActionRequest request = new WebhookWorkflowActionRequest
{
    Url = "https://example.com/webhooks/checkout",
    Headers = new Dictionary<string, string>
    {
        {"Authorization", "70ed20ff-ba31-4ea3-b3ef-772f2be1cbdf"}
    },
    Signature = new WebhookSignature {Method = "HMACSHA256", Key = "public-signing-key"}
};

try
{
    await api.WorkflowsClient().UpdateWorkflowAction("workflow_id", "action_id", request);
}
catch (CheckoutApiException e)
{
    // API error
    string requestId = e.RequestId;
    var statusCode = e.HttpStatusCode;
    IDictionary<string, object> errorDetails = e.ErrorDetails;
}
catch (CheckoutArgumentException e)
{
    // Bad arguments
}
catch (CheckoutAuthorizationException e)
{
    // Invalid authorization
}

File: nas_spec/code_samples/PHP/workflows@{workflowId}@actions@{workflowActionId}/put.php
<?php
//For more information please refer to https://github.com/checkout/checkout-sdk-php

use Checkout\\CheckoutApiException;
use Checkout\\CheckoutAuthorizationException;
use Checkout\\CheckoutSdk;
use Checkout\\Environment;
use Checkout\\OAuthScope;
use Checkout\\Workflows\\Actions\\WebhookSignature;
use Checkout\\Workflows\\Actions\\WebhookWorkflowActionRequest;

//API Keys
$api = CheckoutSdk::builder()->staticKeys()
    ->environment(Environment::sandbox())
    ->secretKey("secret_key")
    ->build();

//OAuth
$api = CheckoutSdk::builder()->oAuth()
    ->clientCredentials("client_id", "client_secret")
    ->scopes([OAuthScope::$Flow])
    ->environment(Environment::sandbox())
    ->build();

$webhookSignature = new WebhookSignature();
$webhookSignature->method = "HMACSHA256";
$webhookSignature->key = "public-signing-key";

$request = new WebhookWorkflowActionRequest();
$request->url = "workflow_name";
$request->signature = "workflow_name";

try {
    $response = $api->getWorkflowsClient()->updateWorkflowAction("workflow_id", "action_id", $request);
} catch (CheckoutApiException $e) {
    // API error
    $error_details = $e->error_details;
    $http_status_code = isset($e->http_metadata) ? $e->http_metadata->getStatusCode() : null;
} catch (CheckoutAuthorizationException $e) {
    // Bad Invalid authorization
}